src/linagora.esn.unifiedinbox/app/services/mailboxes/permissions-service.js

```diff
@@ -45,16 +48,16 @@ angular.module('linagora.esn.unifiedinbox')
       if (!isValidMailbox(mailbox)) {
         return $q.reject(new Error('invalid mailbox provided'));
       }
-      if (!isMyOwnMailbox(mailbox)) { // has permissions to update mailbox sharedWith props
+      if (!isMyOwnMailbox(mailbox)) { // has permissions to update mailbox rights props
         return $q.reject(new Error('Only user ' +
-          (mailbox.namespace && mailbox.namespace.owner || '') +
+          (mailbox.namespace && mailbox.namespace.replace(/^.*\[(.*)\]$/g, '$1') || '') +
```

Comment:
copy/paste detected. This magic regex belongs to a dedicated utility method.

Reply:
I put it in a function.